api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java

```diff
@@ -842,6 +794,41 @@ private CohortDefinition getPatientsWithMonthlyTypeOfDispensation() {
     return cd;
   }
 
+  public SqlCohortDefinition getPatientsWithMonthlyTypeOfDispensation(
+      List<EncounterType> encounterTypes) {
+    SqlCohortDefinition query = new SqlCohortDefinition();
+    StringBuilder queryStr = new StringBuilder();
+    queryStr.append(
+        "select pp.patient_id from (select lst.patient_id, lst.encounter_datetime from (select last_encounter.patient_id, last_encounter.encounter_id, last_encounter.encounter_datetime from (select e.patient_id, e.encounter_datetime, e.encounter_id from encounter e where e.encounter_type in(");
+
+    int i = 0;
+    for (EncounterType et : encounterTypes) {
```

Comment:
I think we need to be consistent in our approaches, all other queries we kept the encounterIds in line not in a loop, I suggest to use the same approach here
@rubailly